### PR TITLE
Added reference documentation policy

### LINKED ISSUE #28: Develop a  reference doc contribution policy
*UPDATE 05/27/2024 (transfer issue from in-toto/docs)*

   - [ ] Specify how and where developers provide reference docs for new implementations. 
   Reference documentation should go with the implementation repo in GitHub, with sources and generation information in a *doc* subdirectory. 
   - [ ] Specify what to document for an implementation. 
   Use the Python Reference Implementation's reference docs to see what needs to be documented. 
   These are generated using source files and a makefile in the implementation's  [*doc* subdirectory](https://github.com/in-toto/in-toto/tree/develop/doc:). 
   - [ ] Add the policy to https://github.com/in-toto/community/blob/main/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by DarikshaAnsari**
Fixes #28

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -37,4 +37,21 @@ The specifics of development practices for a sub-project or implementation are
 detailed in their repositories. For implementations, this includes information
 such as the name of the development branch and any additional checks pertaining
 to that project. For ITEs, the process of submitting a new proposal is described
-in ITE-1.
\ No newline at end of file
+in ITE-1.
+
+## Reference Documentation Policy
+
+For any new implementation, reference documentation should be included in the implementation repository on GitHub. Specifically, the sources and generation information should be located in a `doc` subdirectory within the implementation repository.
```
Comment (lukpueh) [inline CONTRIBUTING.md:44]:
```suggestion
For any new implementation, reference documentation should be included in its repository on GitHub, following documentation best practices specific to the implementation language. 
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -37,4 +37,21 @@ The specifics of development practices for a sub-project or implementation are
...
+
+To ensure comprehensive documentation, refer to the [Python Reference Implementation's reference docs](https://github.com/in-toto/in-toto/tree/develop/doc) as a guideline for the necessary details. These reference docs are generated using source files and a makefile located in the implementation's `doc` subdirectory.
```
Comment (lukpueh) [inline CONTRIBUTING.md:46]:
```suggestion
To ensure comprehensive documentation, refer to the [Python Reference Implementation's reference docs](http://in-toto.readthedocs.io) as a guideline.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -37,4 +37,21 @@ The specifics of development practices for a sub-project or implementation are
...
+#### Example Structure
+```
+implementation-repo/
+├── doc/
+│ ├── source/
+│ ├── Makefile
+│ └── ...
+└── src/
+└── ...
+```
```
Comment (lukpueh) [inline CONTRIBUTING.md:57]:
```suggestion

```

**Review by lukpueh — CHANGES_REQUESTED** [review]
Very nice! IMO we don't need to refer to the docs setup used by the Python in-toto implementation, as It's very Python-specific. But referring to the contents makes sense.

**Comment by lukpueh** [comment]
I think there was a misunderstanding. I specifically suggested to not include Python-specific instructions. If you remove the last commit, we can merge this. :)

**Comment by lukpueh** [comment]
Okay. Did you see my comments in #755? I transferred the issue to in-toto/in-toto and suggested to add Python in-toto docs guidelines to the Python in-toto repo. A PR against https://github.com/in-toto/in-toto/blob/develop/doc/source/CONTRIBUTING.md with the contents from your last commit here would be appreciated.

**Comment by DarikshaAnsari** [comment]
Okay I got your point now i will revert this last commit then.

**Review by lukpueh — APPROVED** [review]
Thanks!

**Pull request merged**